src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,11 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        Optional<User> userOptional = Optional.ofNullable(user);
+        return userOptional.filter(u -> u.getAge() != null)
+                .filter(u -> u.getAge() >= 30)
+                .filter(u -> u.getAge() <= 45)
+                .isPresent();
```

Comment:
Nit: `map()`을 활용해서 처리해보셔도 좋을 것 같습니다.
```suggestion
        return Optional.ofNullable(user)
                .map(User::getAge)
                .filter(age -> age >= 30 && age <= 45)
                .isPresent();
```